clairmeta/dcp_check_am.py

```diff
@@ -116,18 +117,55 @@ def check_assets_am_path(self, am, asset):
         """ AssetMap assets path validation.
 
             Reference :
-                SMPTE 429-9-2014 7.1
+                SMPTE 429-9-2014 7.1, A.1
         """
-        uuid, path, _ = asset
 
-        if path == '':
-            raise CheckException("Empty path for {}".format(uuid))
+        path_errors = []
+        _, path, _ = asset
+
+        path_segments = list(filter(None, path.split('/')))
+        path_segments_count = len(path_segments)
+        if path_segments_count > 10:
+            path_errors.append(">10 path segments: {}".format(path_segments_count))
 
-        if ' ' in path:
-            raise CheckException("Space in path")
+        max_path_seg = max(map(len, path_segments))
+        if max_path_seg > 100:
+            path_errors.append("Path segment >100 characters: {}".format(max_path_seg))
+
+        if len(path) > 100:
+            path_errors.append("Path >100 characters: {}".format(len(path)))
+
+        path_invalid_chars = re.findall(r'[^a-zA-Z0-9._/-]', path)
+        if path_invalid_chars:
+            path_errors.append("Invalid characters in path: {}".format(str(sorted(set(path_invalid_chars)))[1:-1]))
```

Comment:
Why is the [1:-1] needed here ?

Reply:
It is just cosmetic. To remove the leading and trailing brackets in the string representation of a Python list in the output. I thought it may be confusing to a user who is not fluent in Python.